unitylibs/core/workflow/workflow-firefly/action-binder.js

```diff
@@ -225,9 +226,67 @@ export default class ActionBinder {
     if (execute) await execute();
   }
 
-  getSelectedVerbType = () => this.widgetWrap.getAttribute('data-selected-verb');
+  getSelectedVerbType = () => {
+    if (!this.selectedVerbType) {
+      this.selectedVerbType = this.widgetWrap.getAttribute('data-selected-verb');
+    }
+    return this.selectedVerbType;
+  };
+
+  validateInput() {
+    if (this.inputField.value.length === 0 && !this.id) {
+      this.serviceHandler.showErrorToast({ errorToastEl: this.errorToastEl, errorType: '.icon-error-empty-input' }, 'Empty input');
+      return { isValid: false, errorCode: 'empty-input' };
+    }
+    if (this.inputField.value.length > 750) {
+      this.serviceHandler.showErrorToast({ errorToastEl: this.errorToastEl, errorType: '.icon-error-max-length' }, 'Max prompt characters exceeded');
+      return { isValid: false, errorCode: 'max-prompt-characters-exceeded' };
+    }
+    return { isValid: true };
+  }
+
+  async initAnalytics() {
+    if (!this.sendAnalyticsToSplunk && this.workflowCfg.targetCfg.sendSplunkAnalytics) {
+      this.sendAnalyticsToSplunk = (await import(`${getUnityLibs()}/scripts/splunk-analytics.js`)).default;
+    }
+  }
+
+  logAnalyticsinSplunk(eventName, data) {
+    if (this.sendAnalyticsToSplunk) {
+      this.sendAnalyticsToSplunk(eventName, this.workflowCfg.productName, data, `${unityConfig.apiEndPoint}/log`);
+    }
+  }
+
+  logWorkflowStart(eventName, data) {
+    this.logAnalyticsinSplunk(eventName, {
+      ...data,
+      workflowStep: 'start',
+    });
+  }
+
+  logWorkflowComplete(eventName, data) {
+    this.logAnalyticsinSplunk(eventName, {
+      ...data,
+      workflowStep: 'complete',
+    });
+  }
+
+  logErrorAnalytics(eventName, data) {
+    this.logWorkflowComplete(eventName, {
+      ...data,
+      statusCode: -1,
+    });
+  }
+
+  logSuccessAnalytics(eventName, data) {
+    this.logWorkflowComplete(eventName, {
+      ...data,
+      statusCode: 0,
+    });
+  }
```

Comment:
Can we combine them in one function to optimize and remove duplicate code
```suggestion
logAnalytics(eventName, data, options = {}) {
    const { workflowStep, statusCode } = options;
    const logData = { ...data };
    if (workflowStep) logData.workflowStep = workflowStep;
    if (typeof statusCode !== 'undefined') logData.statusCode = statusCode;
    if (this.sendAnalyticsToSplunk) {
      this.sendAnalyticsToSplunk(eventName, this.workflowCfg.productName, logData, `${unityConfig.apiEndPoint}/log`);
    }
  }
}
```

Reply:
Thank you @Ruchika4 for your feedback. 
I do resonate with your concern of having more lines of code. But while it looks like duplicate, but these are separate method calls with varied event data. 
Current implementation looks cleaner and more importantly follows SRP, adhering to clean code principles.
Moreover, I would like to avoid "if" checks (inside the suggested "logAnalytics" method)  as much as possible to reduce the cyclomatic complexity.
But let's connect on this and try to achieve best of both worlds, i.e., less lines of code but still following the clean code principles. Will be happy to know your thoughts.
For now, I have made some attempts to optimize the code lines, please see.

Reply:
Thanks @rohitsahu  for looking into it. I do agree coding practices that you shared. However performance is something that is very important for a.com pages hence file size optimization gets crucial. Regarding cyclomatic complexity suggested code has cyclomatic complexity 4 which comes in range of low CC(1-10) so it should be simple and easy to maintain. We can connect to discuss more on this.

Reply:
I have tried to further update the function to reduce the CC by removing if. Please see if this works as expected for your use case
```
logAnalytics(eventName, data, { workflowStep, statusCode } = {}) {
  const logData = {
    ...data,
    ...(workflowStep && { workflowStep }),
    ...(typeof statusCode !== 'undefined' && { statusCode }),
  };
  this.sendAnalyticsToSplunk?.(
    eventName,
    this.workflowCfg.productName,
    logData,
    `${unityConfig.apiEndPoint}/log`
  );
}
```